incidents/2024-12-10--reviewed-preprints-500.md

```diff
@@ -0,0 +1,67 @@
+* 10:11: Ash provides an update in Slack: "Aisha has locally reverted a server utility function commit, but the error persists."
+* 10:50?: The team identifies that the 500 errors are caused by a request made to the Metrics API, which fails to handle non-standard response codes (e.g., 2xx, 404).
+* ~10:50: Ali runs high-state on Metrics service. Succeeds but doesn't improve things.
+* 10:53: Ash updates the team: "We've identified a Metrics API work completed yesterday may be the root cause. We are working to resolve the issue and ensure such upgrades don’t cause the same impact in the future."
+* 11:00: A fix is committed to the server application to log errors from Metrics API requests instead of throwing them: [Commit: 4dad1f1](https://github.com/elifesciences/enhanced-preprints-server/commit/4dad1f10622070fbefa6b42a3c4bca3dbdeac115).
+* 11:17: The fix is deployed to staging, resolving the 500 errors. Reviewed Preprints pages are visible again without metrics data.
+* 11:22: The fix is deployed to production, resolving the issue for all users. Reviewed Preprints pages are accessible but do not display metrics.
+* 12:32: Ali sends report to Scott of Metrics server issues
+* 13:20: Scott begins investigation. Kernel level error. Restarted Metrics server. Still choppy for 10 minutes and then remained resilient. Increased workers to Metrics server from 4 to 8 (not clear that this has had any impact).
+
+## Contributing factors
+
+* Developers were travelling
+
+## Stabilisation steps
+
+* The team identified and reverted a renovate commit to rule out its involvement. (unrelated, it has been reintroduced)
+* Collaboratively debugged the Metrics API's behaviour and identified the root cause.
+* Committed and deployed a fix to the EPP server application, logging Metrics API errors as soft failures rather than throwing them.
+* Verified the fix on staging and production, ensuring Reviewed Preprints pages were accessible again.
+
+## Impact
+
+* All users attempting to access Reviewed Preprints experienced 500 errors.
+* Reviewed Preprints pages were inaccessible for approximately 5 hours 22 minutes (MTTD: ~2:54 | MTTR: ~2:26 after detection).
+* Metrics data was unavailable on Reviewed Preprints pages after resolution briefly.
+
+## Corrective Actions
+
+*Because we ran out of time in the post-mortem we are going to have to progress on this section asynchronously. If your name is against it please could you be responsible for tidying up the bullet point. Create a ticket, if appropriate and link to the ticket. We are really looking for small changes that can be made that will offer great value.*
+
+*Once you have finished with the action then remove your name from the bullet point.*
+
+* Turn on fastly for EPP/Journal on reviewed preprints (https://github.com/elifesciences/enhanced-preprints-issues/issues/1228)
+* ? More descriptive logs (agree on framework for logs) (Hazal)
+* ? Better monitoring of services (Will/Scott)
```

Comment:
@will-byrne please can you update the corrective action with the link and remove your name now you have created the ticket: https://github.com/elifesciences/enhanced-preprints-issues/issues/1230